### PR TITLE
chore(ci): Bump runner versions

### PR DESCRIPTION
**Pull request by tibetiroka**
**CI/CD/Testing**

Updates the containers for some CI/CD runners.

## Summary
Adds Ubuntu 24 support and drops Ubuntu 22, in line with the existing 'two latest LTS releases' testing methodology.

Drops macOS 12 support and moves to macOS 13, as 12 is expected to be end-of-life before/around the next unstable release (in the next 2-4 months).

Deletes the unused steam.yml workflow file.

Removes a non-standard optimization that breaks apt on Ubuntu 24.

### CODE REVIEW

**Comment by TomGoodIdea** [comment]
There's still a lot of leftover code from the multithread PR.

**Comment by tibetiroka** [comment]
Oh, I force pushed the wrong branch. I was wondering why it was weird...

File: .github/workflows/cd.yaml [hunk 1]
```diff
@@ -27,9 +27,8 @@ jobs:
           show-progress: false
       - name: Install dependencies
         run: |
-          sudo rm /etc/apt/sources.list.d/* && sudo dpkg --clear-avail # Speed up installation and get rid of unwanted lists
           sudo apt-get update
-          sudo apt-get install -y --no-install-recommends libxmu-dev libxi-dev libgl-dev libglu1-mesa-dev libgles2-mesa-dev libwayland-dev libxkbcommon-dev libegl1-mesa-dev
+          sudo apt-get install -y --no-install-recommends libxmu-dev libxi-dev libgl-dev libglu1-mesa-dev libgles2-mesa-dev libwayland-dev libxkbcommon-dev libegl1-mesa-dev fuse
```
Comment (eebop) [inline .github/workflows/cd.yaml:31]: My understanding is that fuse3 is newer & preferred, is fuse required?

Comment (tibetiroka) [reply .github/workflows/cd.yaml:31]: Do they have separate packages now? That's annoying. Will stuff packaged with fuse 3 work with fuse 2 clients?

Comment (eebop) [reply .github/workflows/cd.yaml:31]: I'm pretty sure there's some changes to the .so . I know from experience, I have a app that uses fuse2 and it's a hassle

Comment (tibetiroka) [reply .github/workflows/cd.yaml:31]:
```suggestion
          sudo apt-get install -y --no-install-recommends libxmu-dev libxi-dev libgl-dev libglu1-mesa-dev libgles2-mesa-dev libwayland-dev libxkbcommon-dev libegl1-mesa-dev fuse3
```

File: .github/workflows/cd_release.yaml [hunk 2]
```diff
@@ -21,9 +21,8 @@ jobs:
           show-progress: false
       - name: Install dependencies
         run: |
-          sudo rm /etc/apt/sources.list.d/* && sudo dpkg --clear-avail # Speed up installation and get rid of unwanted lists
           sudo apt-get update
-          sudo apt-get install -y --no-install-recommends libxmu-dev libxi-dev libgl-dev libglu1-mesa-dev libgles2-mesa-dev libwayland-dev libxkbcommon-dev libegl1-mesa-dev
+          sudo apt-get install -y --no-install-recommends libxmu-dev libxi-dev libgl-dev libglu1-mesa-dev libgles2-mesa-dev libwayland-dev libxkbcommon-dev libegl1-mesa-dev fuse
```
Comment (tibetiroka) [inline .github/workflows/cd_release.yaml:25]:
```suggestion
          sudo apt-get install -y --no-install-recommends libxmu-dev libxi-dev libgl-dev libglu1-mesa-dev libgles2-mesa-dev libwayland-dev libxkbcommon-dev libegl1-mesa-dev fuse3
```

File: .github/workflows/cd.yaml (hunk 1 above)
Comment (tibetiroka) [reply .github/workflows/cd.yaml:31]: Couldn't get it to work with fuse 3.

**Review by petervdmeer — APPROVED** [review]

**Comment by petervdmeer** [comment]
> Deletes the unused steam.yml workflow file.

Are you sure it is unused? It was recently updated in https://github.com/endless-sky/endless-sky/pull/10103

**Comment by petervdmeer** [comment]
> > Deletes the unused steam.yml workflow file.
> 
> Are you sure it is unused? It was recently updated in #10103

As discussed on Discord; it is currently indeed unused.

**Pull request merged**